lib/nimble_options/docs.ex

```diff
@@ -75,8 +76,46 @@ defmodule NimbleOptions.Docs do
   end
 
   defp get_default_str(schema) do
-    if Keyword.has_key?(schema, :default) do
-      "The default value is `#{inspect(schema[:default])}`."
+    if Keyword.has_key?(schema, :default),
+      do: "The default value is `#{inspect(schema[:default])}`."
+  end
+
+  defp get_type_str(schema) do
+    case schema[:type] do
+      nil ->
+        nil
+
+      {:custom, _module, _function, _args} ->
+        nil
+
+      {:fun, arity} ->
+        "function of arity #{arity}"
+
+      {:keyword_list, _} ->
+        "keyword list"
+
+      {:in, values} ->
+        "one of " <> Enum.map_join(values, ", ", &get_type_str(type: &1))
```

Comment:
`:in` is for checking membership into an enumerable, not for types. This should rather be:

```suggestion
        "one of #{inspect(values)}"
```